pkg/suggestion/v1alpha3/hyperopt_service.py

```diff
@@ -40,17 +41,52 @@ def GetSuggestions(self, request, context):
         )
 
     def ValidateAlgorithmSettings(self, request, context):
+        is_valid, message = OptimizerConfiguration.validate_algorithm_spec(
+            request.experiment.spec.algorithm)
+        if not is_valid:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details(message)
+            logger.error(message)
         return api_pb2.ValidateAlgorithmSettingsReply()
 
 
-class OptimizerConfiguration(object):
-    def __init__(self, random_state=None):
-        self.random_state = random_state
+class OptimizerConfiguration:
+    __schema_dict = {
+        'tpe': {
+            'gamma': (lambda x: float(x), lambda x: 1 > float(x) > 0),
+            'prior_weight': (lambda x: float(x), lambda x: float(x) > 0),
+            'n_EI_candidates': (lambda x: int(x), lambda x: int(x) > 0),
+            "random_state": (lambda x: int(x), lambda x: x.isdigit()),
```

Comment:
Maybe instead of `x.isdigit()` we can check `int(x) >= 0`, like we did above?

Reply:
@andreyvelich Done.